src/server/dfly_bench.cc

```diff
@@ -645,11 +715,20 @@ void Driver::ParseRESP() {
           CHECK_EQ(parts.size(), 2u);
           uint32_t slot_id;
           CHECK(absl::SimpleAtoi(parts[0], &slot_id));
-          vector<string_view> host_port = absl::StrSplit(parts[1], ':');
-          CHECK_EQ(host_port.size(), 2u);
 
-          // TODO: to support slot migration.
-          LOG_EVERY_T(INFO, 1) << "Moved: " << slot_id << " to " << parts[1];
+          vector<string_view> addr_parts = absl::StrSplit(parts[1], ':');
+          CHECK_EQ(2u, addr_parts.size());
+          string host(addr_parts[0]);
+          char ip_addr[INET6_ADDRSTRLEN];
+          std::error_code ec = fb2::DnsResolve(host, ip_addr);
+          CHECK(!ec) << "Could not resolve " << host << " " << ec;
+          auto address = ::boost::asio::ip::make_address(ip_addr);
+
+          uint32_t port;
+          CHECK(absl::SimpleAtoi(addr_parts[1], &port));
+          CHECK_LT(port, 65536u);
+
+          shard_slots_.MoveSlot(ep_, tcp::endpoint(address, port), slot_id);
```

Comment:
the way I see it, Dragonfly/Redis should respond with a consistent language, so given the fact that our shards are the same during the test, you need to resolve once when the test started (resolution itself is an implementation detail of the Connect function).